content/sips/sip-230.md

```diff
@@ -0,0 +1,99 @@
+* `failureTimestamp(bytes32 id)`: used to get the time at which a breaker was tripped or when the last update was (if the value is stale)
+* `getValues(bytes32 id, uint roundStart, uint roundEnd) returns (uint[] values, uint broken)`: Same as `getValue,` but retrieves a range of values between `roundStart` and `roundEnd`
+* `setValue(bytes32 id, uint value)`: Allows owner to manually set an aggregator price to a static value. Useful for test overrides or defined prices like `sUSD`. Not compatible with an aggregator previously added using `addAggregator`.
+
+#### Circuit Breaker Functionality
+
+Our goal with this system is to prevent large, impossible swings in chainlink price from causing widespread damage to the system, while still not triggering
+unnecessarily when buisness is as usual.
+
+To accomplish this, a simple threshold comparison function is proposed
+
+#### Migration
+
+It will be necessary to migrate the aggregators from the current `ExchangeRates` contract to the new `OracleManager`. Some portions of the deployment
+process will need to be updated to add aggregators to the new manager as well.
+
+Aggregators for `DebtRatio` and `IssuedSynths` will need to be updated to be provisioned to the new `OracleManager` as well
+
+#### Simplified Example Workflows
+
+##### `Exchanger.exchange`
+
+* `Exchanger.exchange`
+  * `ExchangeRates.effectiveValueAndRatesAtRound`
+    * `OracleManager.getValues` -- get range of rates for advanced exchange rate calculations
+  * `ExchangeRates.effectiveValueAndRatesAtRound`
+    * `OracleManager.getValues`
+
+Basically, every call that used to just be against the raw aggregator is now
+
+`ExchangeRatesWithFeeRecAlternatives` and `ExchangeRatesWithDexPricing` will be updated the same way.
+
+##### `Issuer.issueMaxSynths`
+
+* `Issuer.issueMaxSynths`
+  * `Issuer._debtBalanceOfAndTotalDebt`
+    * `Issuer.allNetworksDebtInfo`
+      * `OracleManager.getValue` -- get issued synths oracle
+      * `OracleManager.getValue` -- get issued synth
+    * `ExchangeRates.rateAndInvalid` -- SNX price request
+      * `OracleManager.getValue` -- get SNX price
+  * ...
+
+Similar for Issuer. By doing this, we can mostly remove code for circuit breaking from `Issuer` and take appropropriate actions to protect from oracle accidents by default.
+
+### Configurable Values (Via SCCP)
+
+* `breakerThreshold(bytes32 id)`: defined the factor of change required for the breaker to trip for a given aggregator. If unset, this defaults to `2`
+* `breakerTimeout(uint time)`: number of seconds between calls to `getValue` at which point the breaker will be disabled (i.e. to prevent large gaps in time from trigerring the breaker). Set to 1 to completely disable the breaker. Default value is 86400 seconds--1 day.
```

Comment:
i added it because there is a legitimate case where a price could be very different from the stored value (especially when it comes to issued synths oracle). In practice the only reason this value would come into play is if the synth is both very volitile and not very active, and nobody trades on it for longer than the timeout

Reply:
From discussion: likely unnecessary and unlikely. The timeout would only make sense if it were a very large value (e.g. a month), but if the price was inactive for such a long period, it's also unlikely that it changed by a factor of 3 without any trades through the system. In which case, it should be ok to have false positive. 

This is to avoid having a false negative in a scenario that actually requires breaking, but doesn't trigger because of the timeout.